### ISSUE TITLE
Rendering differences in createEmbeddedView

### ISSUE THREAD

**Issue by digoburigo** [angular/angular#41501]
# 🐞 bug report

### Affected Package
None

### Is this a regression?

I dont' know

### Description

So i'm trying to use a custom logic of portals in Angular to make my application more dynamic,
but when i refresh the page the embedded view don't render properly and this also occurs when
manipulating with conditional rendering like `ngIf`

When wrapping the logic that attaches the embedded view in the template with a `setTimeout` it's
starts working

In stackblitz it's working properly but when doing the same thing in a real project don't work.

## 🔬 Minimal Reproduction

In stackblitz it's working
https://stackblitz.com/edit/angular-ivy-vtppsg?file=src%2Fapp%2Fhome%2Fhome.component.ts
Project that the bug occurs
[teleport.zip](https://github.com/angular/angular/files/6275719/teleport.zip)

## 🔥 Exception or Error
Does not occur any error, it's simply dont render on refresh, change screen sizes and conditional rendering

## 🌍  Your Environment

**Angular Version:**
<pre><code>
...
"@angular/core": "11.2.9",
...
"@angular/cli": "11.2.7",
...
</code></pre>

**Anything else relevant?**
This is all based in this repository: https://github.com/nirkaufman/teleport

**Comment by jelbourn** [comment]
@digoburigo could you clarify more specifically what bug you're reporting here? It's not clear what specifically the difference you're seeing is.

**Comment by digoburigo** [comment]
@jelbourn Yes. In the project that I posted in the issue ```teleport.zip```, when you serve the project you can see that the areas that I use the portal logic don't render properly on first loads, like refreshing the page.

**First Load**
![image](https://user-images.githubusercontent.com/28462822/114239051-ecde5700-995b-11eb-9540-dd1d0a15aa1b.png)

**Changing routes after the app loads**
![image](https://user-images.githubusercontent.com/28462822/114239158-14352400-995c-11eb-9987-5954daedf660.png)

But this problem only happens in a real project. I tried to reproduce the error in Stackblitz but in there this error don't occur.

Hopefully I clarify enough.

**Comment by AndrewKushnir** [comment]
@digoburigo I've looked at the repro and it contains a number of directives and components, so it might be tricky to debug. Could you please update the repro to keep the minimum set of components/directives that are relevant for the repro (ideally 1 component and 1 directive)? Thank you.

**Comment by digoburigo** [comment]
@AndrewKushnir I tried to reduce the code to the maximum and restructure the folders to make it more understandable, but this structure is the minimum so that the bug can be reproduced, if I try to do more than that it will not be possible to show the bug.
Hope this make it possible to debug
[teleport.zip](https://github.com/angular/angular/files/6307809/teleport.zip)

**Comment by AndrewKushnir** [comment]
@digoburigo is the `teleport.zip` has the same code as GitHub repository that you mentioned?

**Comment by digoburigo** [comment]
@AndrewKushnir No, I modify the strucuture to my needs and to show that the bug occurs in this strucuture. In the repository that I mentioned the bug don't occur

**Comment by AndrewKushnir** [comment]
@digoburigo could you please update GitHub repo that you referred to with the most up-to-date version of the repro (we can't work with archives for security reasons)?

**Comment by digoburigo** [comment]
@AndrewKushnir Yes, here it is https://github.com/digoburigo/teleport

**Comment by digoburigo** [comment]
Any update in this topic? Updated to Angular 12 and this error continues.
There is a new branch "ng-12" in the repo to show this.

**Comment by Akxe** [comment]
Here is a [stackblitz](https://stackblitz.com/edit/angular-ivy-aldx8q?file=src%2Fapp%2Fhello.component.ts) with minimal code. All interesting content is in `hello.component.ts` and `app.component.html`. The bug still appears in 12.1.3
